torch/csrc/jit/pybind_utils.h

```diff
@@ -91,9 +103,37 @@ inline IValue toIValue(py::handle input) {
       s.push_back(toIValue(elem));
     }
     return Tuple::create(s);
+  } else if (PyDict_Check(input.ptr())) {
+    // Check to make sure we can generate useful input/output types
+    auto dict = py::cast<py::dict>(input);
+    at::ivalue::UnorderedMap elems;
+    elems.reserve(py::len(dict));
+
+    TypePtr keyType = nullptr;
+    TypePtr valueType = nullptr;
+    for (auto entry : dict) {
+      auto keyIVal = toDictKeyIValue(entry.first);
+      auto valIVal = toIValue(entry.second);
+
+      if (!keyType) {
+        AT_ASSERT(!valueType);
+        keyType = incompleteInferTypeFrom(keyIVal);
+        valueType = attemptToRecoverType(valIVal);
+      } else {
+        if (*keyType != *incompleteInferTypeFrom(keyIVal) ||
+            *valueType != *attemptToRecoverType(valIVal))
```

Comment:
Use `isSubtypeOf()` instead of the equality operator. Requiring exact type equality is too strict because we have multiple Tensor types (that represent various degrees of knowledge we have about the properties of the tensor).